src/main/java/de/qabel/desktop/storage/S3ReadBackend.java

```diff
@@ -0,0 +1,77 @@
+package de.qabel.desktop.storage;
+
+import de.qabel.desktop.exceptions.QblStorageException;
+import de.qabel.desktop.exceptions.QblStorageNotFound;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+class S3ReadBackend extends StorageReadBackend {
+
+	private static final Logger logger = LoggerFactory.getLogger(S3ReadBackend.class.getName());
+
+	String root;
+	private final CloseableHttpClient httpclient;
+
+	S3ReadBackend(String bucket, String prefix) {
+		 this("https://"+bucket+".s3.amazonaws.com/"+prefix);
+	}
+
+	S3ReadBackend(String root) {
+		this.root = root;
+		// Increase max total connection
+		PoolingHttpClientConnectionManager connManager = new PoolingHttpClientConnectionManager();
+		connManager.setMaxTotal(20);
+		// Increase default max connection per route
+		connManager.setDefaultMaxPerRoute(20);
```

Comment:
Could you please explain why 20? Is that something you noticed on testing or read from documentation?
A comment or javadoc would probably make sense.